minitask3/Lewis/follow.py

```diff
@@ -0,0 +1,334 @@
+#!/usr/bin/env python
+'''
+Will beacon a Turtlebot3 Waffle toward green objects or wander around if not.
+'''
+
+from random import uniform
+from math import sqrt, pi as PI
+
+import rospy
+import numpy as np
+from sensor_msgs.msg import Image, LaserScan
+from geometry_msgs.msg import Twist, PoseStamped
+from nav_msgs.msg import Path, Odometry
+from cv2 import cv2
+from cv_bridge import CvBridge, CvBridgeError
+from tf import transformations
+
+FORWARD_SPEED = 0.2
+TURN_SPEED = ((2 * PI) / 360) * 25  # 25 Deg/s
+WALK_DISTANCE = 3.0
+MIN_DISTANCE = 0.3
+
+class Follower():  # pylint: disable=too-many-instance-attributes
+    '''
+    Will beacon a Turtlebot3 Waffle toward green objects or wander around if not.
+    '''
+
+    def __init__(self):
+        rospy.on_shutdown(self.shutdown)
+        self.bridge = CvBridge()
+        self.image_sub = rospy.Subscriber('/camera/rgb/image_raw', Image, self.image_callback)
+        self.twist = Twist()
+        self.vel_pub = rospy.Publisher("/cmd_vel", Twist, queue_size=1)
+
+        # For path plotting
+        self.path_publisher = rospy.Publisher("/path", Path, queue_size=10)
+        self.path = Path()
+        self.plot_subscriber = rospy.Subscriber("odom", Odometry, self.plot_trajectory)
+
+        # For obstacle avoidance
+        self.laser_sub = rospy.Subscriber('/scan', LaserScan, self.scan_callback)
+        self.obstacle = True
+        self.direction = 0
+        self.direction_set = False
+
+        # For odometry and position tracking
+        self.curr_pose = {}
```

Comment:
You can initialise all of these in one line with `self.curr_pose = {'theta': 0.0...` etc.